src/explore-education-statistics-common/src/modules/table-tool/components/LocationFiltersForm.tsx

```diff
@@ -57,44 +57,45 @@ const LocationFiltersForm = ({
 
   const formOptions = useMemo(() => Object.entries(options), [options]);
 
-  // Key options by their value to make future lookups faster/easier.
+  // Key options by their id to make future lookups faster/easier.
   const keyedOptions = useMemo(
     () =>
-      formOptions.reduce<Dictionary<LocationOption>>((acc, [, level]) => {
-        level.options.forEach(option => {
+      mapValues(options, level =>
+        level.options.reduce<Dictionary<LocationOption>>((acc, option) => {
           // There are nested options, so we need
           // to iterate through these instead.
           if (option.options) {
             option.options.forEach(subOption => {
-              acc[subOption.value] = subOption;
+              acc[subOption.id ?? ''] = subOption;
             });
           } else {
-            acc[option.value] = option;
+            acc[option.id ?? ''] = option;
           }
-        });
-
-        return acc;
-      }, {}),
-    [formOptions],
+          return acc;
+        }, {}),
+      ),
+    [options],
   );
 
-  const allOptions = useMemo(() => Object.values(keyedOptions), [keyedOptions]);
-
-  const hasSingleOption = allOptions.length === 1;
+  const hasSingleOption =
+    levelKeys.length === 1 &&
+    Object.keys(keyedOptions[levelKeys[0]]).length === 1;
```

Comment:
Can't get my brain around what we're doing here :joy:  I'm guessing we're checking we've got one key and within that key only one value??

Reply:
Yes that's correct, just added additional comment around that.